### PR TITLE
Remove unneded type checks.

### PR DESCRIPTION
**Pull request by ADmad**
They are covered return types specified for the methods.

### CODE REVIEW

File: src/Core/ServiceProvider.php [hunk 1]
```diff
@@ -44,25 +44,11 @@ abstract class ServiceProvider extends AbstractServiceProvider implements Bootab
...
-            $message = sprintf(
-                'Unexpected container type. Expected `%s` got `%s` instead.',
-                ContainerInterface::class,
-                get_debug_type($container)
-            );
-            throw new RuntimeException($message);
-        }
-
-        return $container;
+        return parent::getContainer();
```
Comment (othercorey) [inline src/Core/ServiceProvider.php:51]: How is this set if it's supposed to be cake's ContainerInterface?

Comment (ADmad) [reply src/Core/ServiceProvider.php:52]: Because Cake's `ContainerInterface` extends League's `ServiceProviderInterface`.

Comment (othercorey) [reply src/Core/ServiceProvider.php:52]: I mean it returns the parent. I assume we set the parent instance somewhere?

Comment (ADmad) [reply src/Core/ServiceProvider.php:52]: We don't need to explicitly set it, it's handled automatically:
https://github.com/thephpleague/container/blob/82a57588c630663d2600f046753b23ab6dcda9b5/src/Container.php#L52
https://github.com/thephpleague/container/blob/82a57588c630663d2600f046753b23ab6dcda9b5/src/ServiceProvider/ServiceProviderAggregate.php#L32

File: src/Core/ServiceProvider.php [hunk 2]
```diff
@@ -44,25 +43,13 @@ abstract class ServiceProvider extends AbstractServiceProvider implements Bootab
     /**
      * Get the container.
      *
-     * This method's actual return type and documented return type differ
-     * because PHP 7.2 doesn't support return type narrowing.
-     *
      * @return \Cake\Core\ContainerInterface
+     * @psalm-suppress MoreSpecificReturnType
```
Comment (ADmad) [inline src/Core/ServiceProvider.php:47]: Don't know why psalm whines about covariant return type when it's supported by the language.

Comment (othercorey) [reply src/Core/ServiceProvider.php]: I think the issue is the method types mismatch. The parent return guarantees only the base class while this return expects the derived class.

The types are allowed but the parent return isn't guaranteed to match.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**